### DISCUSSION TITLE
dark on light theme

### DISCUSSION THREAD

**Discussion by rimzzlabs** [giscus/giscus#820] [Q&A] [answered]
I couldn't figure out what is happening, the body is dark, while the theme is on light mode
![Screen Shot 2022-12-11 at 16 04 09](https://user-images.githubusercontent.com/62492410/206895211-c1c1c3a5-1cbe-49b0-a893-0849783b5a1e.png)

**Comment by rimzzlabs** [comment]
Im using it with Nextjs and here's the component

```tsx
export const GiscusComment = () => {
  const { theme, systemTheme } = useTheme()

  const gcTheme = theme === 'dark' || (theme === 'system' && systemTheme === 'dark') ? 'dark' : 'light'

  return (
    <div className='mt-4 md:mt-8'>
      <Giscus
        lang='en'
        theme={gcTheme}
        emitMetadata='0'
        inputPosition='top'
        repo='rizkimcitra/rizkicitra.dev'
        repoId='R_kgDOGh4MEw'
        category='General'
        categoryId='DIC_kwDOGh4ME84CPxWe'
        mapping='pathname'
        reactionsEnabled='0'
        loading='lazy'
      />
    </div>
  )
}
```

**Comment by laymonage** [comment, marked as answer]
Try updating your `@giscus/react` version to the latest version (v2.2.4): https://github.com/rizkimcitra/rizkicitra.dev/blob/035cfe9f72b75c167e942c3f55c0fa8ef9d7ba83/yarn.lock#L1179-L1184

**Reply by rimzzlabs** [reply]
wow, I'm missing out!
thank you, everything is perfect now
